### ISSUE TITLE
No Truffle v4.x which supports Solc v0.4.26

### ISSUE THREAD

**Issue by barakman** [trufflesuite/truffle#2223]
As you know, Truffle v4.x is tightly-coupled with Solc v0.4.y.
For example:
- Truffle v4.1.14 is designated for Solc v0.4.24.
- Truffle v4.1.15 is designated for Solc v0.4.25.

Truffle v5.x allows the user to configure the desired Solc version (via `truffle-config.js`).
However, since this version of Truffle has switched from Web3 v0.x to Web3 v1.x, anyone upgrading to it would subsequently need to upgrade his/her entire test-suite.

For this reason, anyone who has chosen to stick with Solc v0.4.y (and there are several good reasons to do so at this point), would highly prefer to stick with the corresponding Truffle v4.x.

Up until recently, the latest Truffle v4.x (4.1.15) has relied on the latest Solc v0.4.y (0.4.25).
On April 29, [Solc v0.4.26](https://github.com/ethereum/solidity/releases/tag/v0.4.26) was released, but no corresponding Truffle v4.x has been released since.

Would you be able to release Truffle v4.1.16 which relies on Solc v0.4.26?

Thank you :)

**Comment by fainashalts** [comment]
Thanks for the suggestion @barakman! We'll have a discussion about this and see if we can make this change.

**Comment by CruzMolina** [comment]
Hey @barakman , we will look into releasing this!

**Comment by barakman** [comment]
@CruzMolina:
Thank you very much!
When do you estimate this will be ready?
I'd assume it only takes changing Solc version on your dependencies, though I might be missing some of the big picture.
Thanks :)

**Comment by CruzMolina** [comment]
Hey @barakman , we're looking at releasing this today. It _should_ be a simple matter of bumping the solc-js dep 🤞. The plan is to do a maintenance release if it passes CI.

**Comment by gnidan** [comment]
@barakman could you open the PR to upgrade solc?

**Comment by barakman** [comment]
@gnidan:
To my understanding, the PR should be conducted on the latest v4 release (v4.1.15).
I have no idea how to open a PR on an older release (let alone the fact that there is no way to reach this release from your GitHub).
Thanks.

**Comment by barakman** [comment]
@gnidan:
Hi, is this release due any time soon?

**Comment by gnidan** [comment]
Apologies for the delay @barakman. I've had to lose focus here with the upcoming TruffleCon and our team retreat the week after. There should be a branch `4.x` or something similarly named, to target for the upgrade. There is risk since we haven't done these v4 patches very much, so the release will realistically have to wait until we're back to business as usual, the week of the 12th.

Thanks for understanding!

**Comment by barakman** [comment]
@gnidan: Thanks for the quick update.

The only branch that I've found with a similar name is [v4](https://github.com/trufflesuite/truffle/tree/v4), but it's located under your *stale-branches* list.

Is that the one that you want me to open a PR in?

**Comment by gnidan** [comment]
Yep, that's the one!

**Comment by barakman** [comment]
@gnidan:
I cannot do a PR.
Git response: `Permission to trufflesuite/truffle.git denied to barakman`

**Comment by gnidan** [comment]
Oh, you need to fork first.

**Comment by barakman** [comment]
@gnidan:
```
git.exe push --progress "origin" v4.1.16

remote: Permission to trufflesuite/truffle.git denied to barakman.
fatal: unable to access 'https://github.com/trufflesuite/truffle.git/': The requested URL returned error: 403
```

**Comment by CruzMolina** [comment]
@barakman did you run `yarn add solc@0.4.26` when bumping the solc versions?

**Comment by barakman** [comment]
@CruzMolina:
No, I didn't. Sorry, I wasn't aware of that.
I cannot run it on my system:
```
error C:\Users\Temp\Desktop\truffle\node_modules\sha3: Command failed.
Exit code: 1
Command: node-gyp rebuild
Arguments:
Directory: C:\Users\Temp\Desktop\truffle\node_modules\sha3
Output:
gyp info it worked if it ends with ok
gyp info using node-gyp@3.8.0
gyp info using node@10.16.0 | win32 | x64
gyp ERR! configure error
gyp ERR! stack Error: Command failed: C:\Users\Temp\Anaconda3\python.EXE -c import sys; print "%s.%s.%s" % sys.version_info[:3];
gyp ERR! stack   File "<string>", line 1
gyp ERR! stack     import sys; print "%s.%s.%s" % sys.version_info[:3];
gyp ERR! stack                                ^
gyp ERR! stack SyntaxError: invalid syntax
gyp ERR! stack
gyp ERR! stack     at ChildProcess.exithandler (child_process.js:294:12)
gyp ERR! stack     at ChildProcess.emit (events.js:198:13)
gyp ERR! stack     at maybeClose (internal/child_process.js:982:16)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:259:5)
gyp ERR! System Windows_NT 10.0.17134
gyp ERR! command "C:\\Program Files\\nodejs\\node.exe" "C:\\Users\\Temp\\Desktop\\truffle\\node_modules\\node-gyp\\bin\\node-gyp.js" "rebuild"
gyp ERR! cwd C:\Users\Temp\Desktop\truffle\node_modules\sha3
```
BTW, you might wanna add `yarn` to your `package.json` file.
Thanks

**Comment by eggplantzzz** [comment]
This has been resolved with the release of Truffle v4.1.16
https://github.com/trufflesuite/truffle/releases/tag/v4.1.16
